### ISSUE TITLE
Compiling htrack on MS Visual Studio

### ISSUE THREAD

**Issue by Scherady** [OpenGP/htrack#4]
Dear all,
Thank you for making htrack available. I am trying to compile it in MS Visual Studio. 
In htrack_qt app I reached the point where the window with the hand model appears but I have not succeeded to get the camera stream. I have tried with both Intel Creative DepthSense and RealSense but no success until now, however the application helloworld_realsense seams to work with RealSense camera.
1. Please let me know what I should do to make the make the camera work in tracking mode.
I have also other questions:
2. The binary files P, Sigma and Limits load matrixes with cols = 0. Temporarily to avoid crashing I artificially set cols = 2; what is the problem in your opinion?
3. What are the arguments filename_data and filename_soln for main in htrack_qt?
4. Q_INIT_RESOURCE(shaders) requires the external "qInitResources_shaders", in which library do I find it?
Thank you in advance for your help.

**Comment by tvandenzegel** [comment]
Hi,
1. For htrack_qt app: Did you define the correct (and removed the unused ) preprocessor definitions? See "main.cpp" to make sure the correct one is enabled.

Also don't forget to wear a blue wrist band! :) Maybe it is just that! :)

2. The application only works in 64bit. If you run in 32bit then indeed you encounter bugs like failing to load the matrices (the column and row  index are saved in a 64bit integer).
But actually I assume you are already compile in 64bit, because the needed CUDA libraries are 64bit only.
3. I think you can just leave them empty.





2. Make sure you build in 64bit, there is a

**Comment by Scherady** [comment]
1. What are actually the correct preprocessor definitions, I am looking for them, are they documented somewhere and I missed them?

2. The camera LED and IR illumination do not go on at all, where as in helloworld_realsense they do.

3. Please confirm that the camera for htrack_qt app is actually realsense and not depthsense (same as helloworld_realsense app).

Thank you

**Comment by ataiya** [comment]
The numbering above is somewhat confusing. 
**You'd submit separate tickets for different questions!!**

1) if you start htrack_atb it's already in tracking mode directly

2) yes you need 64 bits compile as noted in the homepage, we'll add a check for it (but in visual studio this might be disabled by the IDE)

3) Please use **htrack_atb**. I left the Qt application in, but that's mostly for recording and re-tracking sequences. Also note on windows the performance of the tracker will be *much* worse. You'll only be able to run at around 40FPS and this will have a visible effect on tracking! You can use the _qt application and you can leave the arguments empty.

4) That's out-of-the-box part of Qt5!

---- 

1-bis) [preprocessor definitions]: none

2-bis) [LED and IR illumination do not go on at all]: check in the main.cpp files what sensor you are instantiating

3-bis) htrack_qt: you should not use it :)

**Comment by Scherady** [comment]
Sorry for the confusion created by double posting.
1. I used htrack_atb but it seams to run only with Kinect, is it the case?
2. I am still not sure which camera is best on Windows.
3. Can htrack run on Mac?
4. Do you recommend to run hrack on Linux rather?
Thank you

**Comment by ataiya** [comment]
1) This is the demo application, it should be executed with SoftKinetic (check the main.cpp in the htrack_atb to find out more on your own)
2) On windows... you only get about 45FPS on a top-end machine, so windows is not really the platform. It *can* compile/run on it, but tracking performance is suboptimal
3) Can run, same problems as windows (OS interfering with CUDA scheduling)
4) Yes, I do *STRONGLY* recommend linux as mentioned on the homepage

No problem

**Comment by Scherady** [comment]
I managed to compile using SoftKinect, and as you said I realised that the performance was insufficient under Windows.
Now that I installed Ubuntu 15.04, 
1. what C++ IDE do you recommend? 
2. what other modules/plugins do I need?
3. what makefile recommendations do you suggest?
Thank you

**Comment by ataiya** [comment]
1. We mostly QtCreator
2. Plugins for what?
3. I just generate normal Makefiles

In linux/mac if from the terminal you execute this:
    
    ~/Developer/htrack: qtcreator CMakeLists.txt  

Qtcreator will open with a CMake setup window that generates makefiles for you and hooks them up to the IDE.

**Comment by Scherady** [comment]
Hi,
I managed to run CMake without errors.

(1) I am getting the following CUDA warnings and errors which I have not succeeded to resolve:
*************************************************************************
11:21:30: Running steps for project htrack...
11:21:30: Starting: "/usr/bin/make" 
[  1%] Building NVCC (Device) object cudax/CMakeFiles/cudax.dir//./cudax_generated_kernel.cu.o
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
In file included from /home/scherady/Scherady/htrack-master/cudax/../cudax/kernel.h:12:0,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:6:
/home/scherady/Scherady/htrack-master/cudax/../cudax/cuda_glm.h:3:0: warning: "CUDA_VERSION" redefined
 #define CUDA_VERSION 7000
 ^
In file included from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/compiler.h:25:0,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config/config.h:26,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/detail/config.h:22,
                 from /usr/lib/nvidia-cuda-toolkit/include/thrust/host_vector.h:24,
                 from /home/scherady/Scherady/htrack-master/cudax/kernel.cu:3:
/usr/include/cuda.h:158:0: note: this is the location of the previous definition
 #define CUDA_VERSION 6050
 ^
/usr/include/c++/4.9/bits/alloc_traits.h(248): error: expected a ">"
          detected during instantiation of type "std::allocator_traits<_Alloc>::__has_construct<_Tp, _Args...>" 
(251): here

/usr/include/c++/4.9/type_traits(130): error: not a class or struct name
          detected during:
            instantiation of class "std::__and_<_B1> [with _B1=<error-type>]" 
(1957): here
            instantiation of type "std::_Require<<error-type>>" 
/usr/include/c++/4.9/bits/alloc_traits.h(251): here

/usr/include/c++/4.9/type_traits(1957): error: class "std::__and_<<error-type>>" has no member "value"
          detected during instantiation of type "std::_Require<<error-type>>" 
/usr/include/c++/4.9/bits/alloc_traits.h(251): here

/usr/include/c++/4.9/type_traits(1957): error: class "std::enable_if<<error-constant>, void>" has no member "type"
          detected during instantiation of type "std::_Require<<error-type>>" 
/usr/include/c++/4.9/bits/alloc_traits.h(251): here

4 errors detected in the compilation of "/tmp/tmpxft_00002131_00000000-15_kernel.compute_50.cpp1.ii".
CMake Error at cudax_generated_kernel.cu.o.cmake:260 (message):
  Error generating file
  /home/scherady/Scherady/build-htrack-master-Desktop-Default/cudax/CMakeFiles/cudax.dir//./cudax_generated_kernel.cu.o


cudax/CMakeFiles/cudax.dir/build.make:760: recipe for target 'cudax/CMakeFiles/cudax.dir/./cudax_generated_kernel.cu.o' failed
CMakeFiles/Makefile2:122: recipe for target 'cudax/CMakeFiles/cudax.dir/all' failed
Makefile:76: recipe for target 'all' failed
make[2]: *** [cudax/CMakeFiles/cudax.dir/./cudax_generated_kernel.cu.o] Error 1
make[1]: *** [cudax/CMakeFiles/cudax.dir/all] Error 2
make: *** [all] Error 2
11:21:35: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project htrack (kit: Desktop)
When executing step 'Make'
11:21:35: Elapsed time: 00:04.
***************************
Please help.

(2) Which sensor should I select ?
******************************************************************
#--- Choose a sensor

SET(SENSOR_DEPTHSENSEGRABBER 0 CACHE BOOL "Use SoftKinetic with DepthSenseGrabber")
SET(SENSOR_SOFTKIN 0 CACHE BOOL "Use SoftKinetic with SoftKin")
SET(SENSOR_REALSENSE 0 CACHE BOOL "Use Intel RealSense")
SET(SENSOR_OPENNI 0 CACHE BOOL "Use PrimeSense Carmine or ASUS Xtion with OpenNI")

SET(SENSOR_COUNT 0)
*****************************
Thank you

**Comment by ataiya** [comment]
1) simple: update to CUDA 7
2) SENSOR_DEPTHSENSEGRABBER (which is the Senz3D sensor as mentioned on homepage). Other sensors would "run", but there are some magic numbers that are set for the Senz3D in the code.

**Comment by Scherady** [comment]
Hi,
I am getting this error now:
**********************************
13:04:49: Running steps for project htrack...
13:04:49: Starting: "/usr/bin/make" 
[  3%] Built target cudax
[ 85%] Built target htracksdk
[ 87%] Building CXX object apps/htrack_atb/CMakeFiles/htrack_atb.dir/main.cpp.o
g++-4.9: error: CMakeFiles/htrack_atb.dir/main.cpp.o: No such file or directory
apps/htrack_atb/CMakeFiles/htrack_atb.dir/build.make:54: recipe for target 'apps/htrack_atb/CMakeFiles/htrack_atb.dir/main.cpp.o' failed
CMakeFiles/Makefile2:286: recipe for target 'apps/htrack_atb/CMakeFiles/htrack_atb.dir/all' failed
Makefile:76: recipe for target 'all' failed
make[2]: *** [apps/htrack_atb/CMakeFiles/htrack_atb.dir/main.cpp.o] Error 1
make[1]: *** [apps/htrack_atb/CMakeFiles/htrack_atb.dir/all] Error 2
make: *** [all] Error 2
13:04:49: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project htrack (kit: Desktop)
When executing step 'Make'
13:04:49: Elapsed time: 00:00.
**********************************
Please advice, thank you.

**Comment by ataiya** [comment]
It's hard to say from only the make log. Start clean and post a copy of the terminal text with the **exact commands** you execute (starting from "git clone"). It sounds to me you are not doing an out-of-source build (shadow build).

**Comment by Scherady** [comment]
Hi,
1. I downloaded the content of OpenGPhtrack as a zip file, I did not use git clone.
2. In termial from htrack-master directory I used the command: qtcreator CMakeLists.txt 
3. In Qt Creator IDE I iteratively run cmake and resolved missing libraries one by one.
4. One possible origin of this error could be that I made a change in ConfigureCompiler.cmake; changed g++-4.8 to g++-4.9 as it is the version installed in my computer:
*****************************************
    message(STATUS "Compiler(Linux): g++-4.9")
    set(CMAKE_CXX_COMPILER "/usr/bin/g++-4.9")
*****************************************
Thank you

**Comment by Scherady** [comment]
Hi,
while still trying to resolve these compilation errors with your help, I added a new issue with more substantial questions.
[Tracking all ten fingers of both hands](https://github.com/OpenGP/htrack/issues/8)
Thank you for your assistance.

**Comment by ataiya** [comment]
Alright, sure, you don't need to use the 'git clone', and the procedure you followed seems correct.
I don't think 4.9 would cause issues (it's so close to 4.8). That's unless CUDA specifically requires 4.8?

(please edit your issue to include the error that has been generated, thanks)

**Comment by Scherady** [comment]
Hi,
I have included a few days ago in this issue the generated error,  please let me know what else I should include.
Thank you.

**Comment by ataiya** [comment]
That error looks very odd. Where is your build folder? 
Is it outside your source as in a [shadow build](http://doc.qt.io/qtcreator/creator-glossary.html#glossary-shadow-build)?

Also, have you succeeded to compile all the "helloworld_*" correctly?

**Comment by Scherady** [comment]
1) Qt has created a build directory:

**************************************************
-- Configuring done
-- Generating done
-- Build files have been written to: /home/brian/Scherady/build-htrack-master-Desktop-Default

**************************************************
In this directory there is directory:
apps/htrack_atb/CMakeFiles/htrack_atb.dir/
but not the file:
main.cpp.o

It contains a build.make file among others

2) How to compile separately the helloworld_*s ? It is not transparent to me. The apps visible in Qt IDE are only htrack_atb and htrack_qt.

Thank you

**Comment by ataiya** [comment]
1) Odd, Odd... Odd. Are you in Linux now? Can you compile from command line?
2) you can simply open the CMakeLists.txt file in the "htrack/apps/" folder

**Comment by Scherady** [comment]
1) :) Solved !
The reason was that I had SET(SENSOR_COUNT 1) it should be 0. 

2) OK

3) I am getting this error now:

********************************
Starting /home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb...
/home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb: error while loading shared libraries: libAntTweakBar.so.1: cannot open shared object file: No such file or directory
/home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb exited with code 127

********************************
Where should I put libAntTweakBar.so.1?

4) Have you received my e-mail?

Thank you for your help.

**Comment by ataiya** [comment]
1) that makes sense why I didn't recognize the error... It's in a pull-request part of the code :)
3) " invalid device function" I think this used to happen when you use the wrong "compute" in the CUDA settings. Are you using a >GTX980 as mentioned in the readme? You can enable the other cards support in the ConfigureCuda script.
4) yes

**Comment by ataiya** [comment]
I don't quite understand why the error in my email notification is completely different
```
Starting /home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb...
/home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb: error while loading shared libraries: libAntTweakBar.so.1: cannot open shared object file: No such file or directory
/home/brian/Scherady/build-htrack-Desktop-Default/htrack_atb exited with code 127
```

**Comment by Scherady** [comment]
1) This is an old notification, as I solved this error,  to save time and unnecessary efforts, I replaced it by the next one I couldn't solve by editing my post.

2) Now I reached the point where I get the window with the blue hand but when I press LIVE the program crashes:
*****************************************************************************

Starting /home/brian/Scherady/build-htrack-Desktop-Default/htrack_qt...
htrack starting 
--Execution path:  /home/brian/Scherady/build-htrack-Desktop-Default 
OpenGL_viewer::initializeGL() OpenGL 3 . 2 
Reading Wristband Colors from: /home/brian/Scherady/build-htrack-Desktop-Default/wristband.txt
  hsv_min: [94, 111, 37, 0]
  hsv_max: [120, 255, 255, 0]
Starting up CUBLAS
Init cuda: 320 240
Executed [Kernel Initialization] in [1.1113ms]
Loading PCA data from: /home/brian/Scherady/build-htrack-Desktop-Default/PoseSpace_PCA/
Calibration::autoload()
Loaded calibrated model from /home/brian/Scherady/build-htrack-Desktop-Default/default.calib
SensorDepthSenseGrabber::initialize()
Found 3 nodes
Waiting for data...  0 
OpenCV Error: Unspecified error (The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Carbon support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script) in cvDestroyWindow, file /home/brian/opencv/modules/highgui/src/window.cpp, line 489
terminate called after throwing an instance of 'cv::Exception'
  what():  /home/brian/opencv/modules/highgui/src/window.cpp:489: error: (-2) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Carbon support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function cvDestroyWindow

The program has unexpectedly finished.
/home/brian/Scherady/build-htrack-Desktop-Default/htrack_qt crashe

****************************************************************************
I installed libgtk2.0-dev and pkg-config was installed but it did not help.

In parallel I managed to get DepthSense Viewer working.

Thank you for your help.

**Comment by ataiya** [comment]
This seems to be a raw OpenCV error.
**Have you tried the helloworld_opencv?**

**Comment by Scherady** [comment]
Trying  helloworld_opencv I get:
*********************************************************
/home/brian/Scherady/htrack-1/apps/helloworld_cudagl/main.cpp:102: error: 'timer' was not declared in this scope
             TIMED_SCOPE(timer,"offscreendraw");
                         ^
*********************************************************
Thank you

**Comment by ataiya** [comment]
Just comment it out?

**Comment by Scherady** [comment]
Next error (without debugging efforts :) ):
****************************************************************

07:28:57: Running steps for project htrack-helloworlds...
07:28:57: Starting: "/usr/bin/make" 
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- Found ANTTWEAKBAR: /usr/local/lib/libAntTweakBar.so
-- AntTweakBar: /usr/local/lib/libAntTweakBar.so
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- GLEW_LIBRARIES: /usr/lib/x86_64-linux-gnu/libGLEW.so
-- HOSTNAME FFTT
-- CUDA_PROPAGATE_HOST_FLAGS: False
-- CUDA_HOST_COMPILER: /usr/bin/cc
-- CUDA_NVCC_FLAGS:  -gencode arch=compute_20,code=sm_20 -gencode arch=compute_30,code=sm_30 -gencode arch=compute_32,code=sm_32 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -DNDEBUG -std c++11 -O3 -use_fast_math
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- HOSTNAME FFTT
-- CUDA_PROPAGATE_HOST_FLAGS: False
-- CUDA_HOST_COMPILER: /usr/bin/cc
-- CUDA_NVCC_FLAGS:  -gencode arch=compute_20,code=sm_20 -gencode arch=compute_30,code=sm_30 -gencode arch=compute_32,code=sm_32 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -DNDEBUG -std c++11 -O3 -use_fast_math
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- HOSTNAME FFTT
-- CUDA_PROPAGATE_HOST_FLAGS: False
-- CUDA_HOST_COMPILER: /usr/bin/cc
-- CUDA_NVCC_FLAGS:  -gencode arch=compute_20,code=sm_20 -gencode arch=compute_30,code=sm_30 -gencode arch=compute_32,code=sm_32 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -DNDEBUG -std c++11 -O3 -use_fast_math
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- CMAKE_BUILD_TYPE Release
-- Compiler(Linux): g++-4.9
-- Found OpenNI: /home/brian/Scherady/htrack-1/openni/lib/libOpenNI2.so
-- !!!WARNING Qt5: resorting to default Qt5 paths (might fail): /usr/local/qt/5.3/gcc_64/lib/cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /home/brian/Scherady/htrack-1/build-apps-Desktop-Default
[  7%] Built target helloworld
[ 14%] Built target helloworld_opencv
[ 21%] Built target helloworld_qtopengl
[ 35%] Built target helloworld_atb
[ 42%] Built target helloworld_cudagl
Linking CXX static library libcudax.a
[ 50%] Built target cudax
Linking CXX executable helloworld_cublas_perfomance
[ 57%] Built target helloworld_cublas_perfomance
Linking CXX executable helloworld_cublas_simple
[ 64%] Built target helloworld_cublas_simple
Linking CXX executable helloworld_thrust
[ 71%] Built target helloworld_thrust
[ 78%] Building CXX object helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/main.cpp.o
/home/brian/Scherady/htrack-1/apps/helloworld_qglviewer/main.cpp:2:33: fatal error: QGLViewer/qglviewer.h: No such file or directory
 #include <QGLViewer/qglviewer.h>
                                 ^
compilation terminated.
helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/build.make:54: recipe for target 'helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/main.cpp.o' failed
make[2]: *** [helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/main.cpp.o] Error 1
make[1]: *** [helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/all] Error 2
make: *** [all] Error 2
CMakeFiles/Makefile2:514: recipe for target 'helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/all' failed
Makefile:76: recipe for target 'all' failed
07:28:59: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project htrack-helloworlds (kit: Desktop)
When executing step 'Make'
07:28:59: Elapsed time: 00:02.

*******************************************************************
Thank you

**Comment by ataiya** [comment]
qglviewer is only for the htrack_qt version. It's quite clear from the error that you simply don't have it installed. But you don't need it anyway if you use the anttweakbar version

**Comment by Scherady** [comment]
1) what is the difference between htrack_qt and htrach_atb ?
2) Now I am trying to run helloworld_opencv to investigate the OpenCV raw error, for that I need qglviewer (I think), I just installed it but there are other issues (it took me long to solve these type of problems under MS-Windows; but I have only a few weeks of experience with Linux): 

************************************************
11:02:01: Running steps for project htrack-helloworlds...
11:02:01: Starting: "/usr/bin/make" 
[  7%] Built target helloworld
[ 14%] Built target helloworld_opencv
[ 21%] Built target helloworld_qtopengl
[ 35%] Built target helloworld_atb
[ 42%] Built target helloworld_cudagl
[ 50%] Built target cudax
[ 57%] Built target helloworld_cublas_perfomance
[ 64%] Built target helloworld_cublas_simple
[ 71%] Built target helloworld_thrust
[ 78%] Building CXX object helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/main.cpp.o
Linking CXX executable helloworld_qglviewer
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `Viewer::~Viewer()':
main.cpp:(.text._ZN6ViewerD2Ev[_ZN6ViewerD5Ev]+0x19): undefined reference to `QGLViewer::~QGLViewer()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `Viewer::~Viewer()':
main.cpp:(.text._ZN6ViewerD0Ev[_ZN6ViewerD5Ev]+0x1d): undefined reference to `QGLViewer::~QGLViewer()'
helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/build.make:103: recipe for target 'helloworld_qglviewer/helloworld_qglviewer' failed
CMakeFiles/Makefile2:514: recipe for target 'helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/all' failed
Makefile:76: recipe for target 'all' failed
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `QGLViewer::helpString() const':
main.cpp:(.text._ZNK9QGLViewer10helpStringEv[_ZNK9QGLViewer10helpStringEv]+0x3): undefined reference to `QGLViewer::staticMetaObject'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `QGLViewer::animate()':
main.cpp:(.text._ZN9QGLViewer7animateEv[_ZN9QGLViewer7animateEv]+0x1): undefined reference to `QGLViewer::animateNeeded()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `QGLViewer::setSceneCenter(qglviewer::Vec const&)':
main.cpp:(.text._ZN9QGLViewer14setSceneCenterERKN9qglviewer3VecE[_ZN9QGLViewer14setSceneCenterERKN9qglviewer3VecE]+0x5): undefined reference to `qglviewer::Camera::setSceneCenter(qglviewer::Vec const&)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `QGLViewer::setSceneRadius(float)':
main.cpp:(.text._ZN9QGLViewer14setSceneRadiusEf[_ZN9QGLViewer14setSceneRadiusEf]+0x5): undefined reference to `qglviewer::Camera::setSceneRadius(float)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o: In function `main':
main.cpp:(.text.startup+0x7f): undefined reference to `QGLViewer::QGLViewer(QWidget*, QGLWidget const*, QFlags<Qt::WindowType>)'
main.cpp:(.text.startup+0xf6): undefined reference to `QGLViewer::~QGLViewer()'
main.cpp:(.text.startup+0x15a): undefined reference to `QGLViewer::~QGLViewer()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTI6Viewer[_ZTI6Viewer]+0x10): undefined reference to `typeinfo for QGLViewer'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x10): undefined reference to `QGLViewer::metaObject() const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x18): undefined reference to `QGLViewer::qt_metacast(char const*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x20): undefined reference to `QGLViewer::qt_metacall(QMetaObject::Call, int, void**)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x48): undefined reference to `QGLViewer::timerEvent(QTimerEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xa8): undefined reference to `QGLViewer::mousePressEvent(QMouseEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xb0): undefined reference to `QGLViewer::mouseReleaseEvent(QMouseEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xb8): undefined reference to `QGLViewer::mouseDoubleClickEvent(QMouseEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xc0): undefined reference to `QGLViewer::mouseMoveEvent(QMouseEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xc8): undefined reference to `QGLViewer::wheelEvent(QWheelEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xd0): undefined reference to `QGLViewer::keyPressEvent(QKeyEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0xd8): undefined reference to `QGLViewer::keyReleaseEvent(QKeyEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x118): undefined reference to `QGLViewer::closeEvent(QCloseEvent*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x1c0): undefined reference to `QGLViewer::initializeGL()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x1c8): undefined reference to `QGLViewer::resizeGL(int, int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x1d0): undefined reference to `QGLViewer::paintGL()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x210): undefined reference to `QGLViewer::startScreenCoordinatesSystem(bool) const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x218): undefined reference to `QGLViewer::stopScreenCoordinatesSystem() const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x220): undefined reference to `QGLViewer::drawLight(unsigned int, float) const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x228): undefined reference to `QGLViewer::startAnimation()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x230): undefined reference to `QGLViewer::stopAnimation()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x248): undefined reference to `QGLViewer::mouseString() const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x250): undefined reference to `QGLViewer::keyboardString() const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x268): undefined reference to `QGLViewer::help()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x270): undefined reference to `QGLViewer::aboutQGLViewer()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x280): undefined reference to `QGLViewer::preDraw()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x288): undefined reference to `QGLViewer::preDrawStereo(bool)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x298): undefined reference to `QGLViewer::fastDraw()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2a0): undefined reference to `QGLViewer::postDraw()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2a8): undefined reference to `QGLViewer::select(QMouseEvent const*)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2b0): undefined reference to `QGLViewer::select(QPoint const&)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2b8): undefined reference to `QGLViewer::beginSelection(QPoint const&)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2c8): undefined reference to `QGLViewer::endSelection(QPoint const&)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2d8): undefined reference to `QGLViewer::setKeyFrameKey(int, int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2e0): undefined reference to `QGLViewer::setPlayKeyFramePathStateKey(int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2e8): undefined reference to `QGLViewer::setPlayPathStateKey(int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2f0): undefined reference to `QGLViewer::setAddKeyFrameStateKey(int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x2f8): undefined reference to `QGLViewer::setPathKey(int, int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x300): undefined reference to `QGLViewer::setPlayPathKeyboardModifiers(QFlags<Qt::KeyboardModifier>)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x308): undefined reference to `QGLViewer::setAddKeyFrameKeyboardModifiers(QFlags<Qt::KeyboardModifier>)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x310): undefined reference to `QGLViewer::domElement(QString const&, QDomDocument&) const'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x318): undefined reference to `QGLViewer::initFromDOMElement(QDomElement const&)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x320): undefined reference to `QGLViewer::saveStateToFile()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x328): undefined reference to `QGLViewer::restoreStateFromFile()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x330): undefined reference to `QGLViewer::setVisualHintsMask(int, int)'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x338): undefined reference to `QGLViewer::drawVisualHints()'
CMakeFiles/helloworld_qglviewer.dir/main.cpp.o:(.data.rel.ro._ZTV6Viewer[_ZTV6Viewer]+0x340): undefined reference to `QGLViewer::resetVisualHints()'
collect2: error: ld returned 1 exit status
make[2]: *** [helloworld_qglviewer/helloworld_qglviewer] Error 1
make[1]: *** [helloworld_qglviewer/CMakeFiles/helloworld_qglviewer.dir/all] Error 2
make: *** [all] Error 2
11:02:03: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project htrack-helloworlds (kit: Desktop)
When executing step 'Make'
11:02:03: Elapsed time: 00:03.

**************************************************

Thank you

**Comment by Scherady** [comment]
OK !  :)
I commented out helloworld_qglviewer and compiled the the other helloworld*s. They all run OK except helloworld_opencv that gives the error:

**************************************************
Starting /home/brian/Scherady/htrack-1/build-apps-Desktop-Default/helloworld_opencv/helloworld_opencv...
image size: 512 512
OpenCV Error: Unspecified error (The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Carbon support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script) in cvShowImage, file /home/brian/opencv/modules/highgui/src/window.cpp, line 501
terminate called after throwing an instance of 'cv::Exception'
  what():  /home/brian/opencv/modules/highgui/src/window.cpp:501: error: (-2) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Carbon support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function cvShowImage

The program has unexpectedly finished.
/home/brian/Scherady/htrack-1/build-apps-Desktop-Default/helloworld_opencv/helloworld_opencv crashed

*************************************************
Thank you

**Comment by ataiya** [comment]
If the helloworld doesn't run, that means your problem is to be solved on the OpenCV forums :)
Open the file in the folder, as you can see we are executing very little code!!
Note it's OpenCV 2.4 we use, not the latest 3.0 (released after we published)
